src/App.tsx

```diff
@@ -1,61 +1,150 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+}
+
+interface User {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+}
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const users = [...usersFromServer];
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState<null | number>(null);
+  const [myTodos, setMyTodos] = useState(todos);
+  const [isWrongUser, setIsWrongUser] = useState(false);
+  const [isEmptyTitle, setIsEmptyTitle] = useState(false);
+
+  const addTodo = () => {
+    const arr = myTodos.sort(
+      (todo1, todo2) => todo1.id - todo2.id,
+    );
+
+    const largestId = arr[arr.length - 1].id;
+
+    if (userId) {
+      setMyTodos(state => (
+        [...state, {
+          id: largestId + 1,
+          title,
+          completed: false,
+          userId,
+          user: getUser(userId),
+        }]
+      ));
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (title !== '' && userId === null) {
+      setIsWrongUser(true);
+    } else if (!isEmptyTitle && userId !== null && title.trim() !== '') {
+      addTodo();
+      setTitle('');
+      setUserId(null);
+    } else if (title === '' || userId === null) {
+      setIsEmptyTitle(true);
+      setIsWrongUser(true);
+    } else if (title.trim() === '') {
+      setIsEmptyTitle(true);
+    }
```

Comment:
Generally try to avoid if/else, else if
Check this 
https://www.linkedin.com/pulse/avoid-if-else-hell-javascript-afsal-m-h/